tests/preprocessing/test_data.py

```diff
@@ -471,3 +471,109 @@ def test_inverse_transform(self):
         assert_eq_df(df, enc.inverse_transform(enc.transform(df)))
         assert_eq_df(df, enc.inverse_transform(enc.transform(df).values))
         assert_eq_df(df, enc.inverse_transform(enc.transform(df).values))
+
+
+class TestPolynomialFeatures:
+    def test_basic(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        a.fit(X)
+        b.fit(X.compute())
+        assert_estimator_equal(a._transformer, b)
+
+    def test_input_types(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        assert_estimator_equal(a.fit(df), a.fit(df.compute()))
+        assert_estimator_equal(a.fit(df), a.fit(df.compute().values))
+        assert_estimator_equal(a.fit(df.values), a.fit(df.compute().values))
+        assert_estimator_equal(a.fit(df), b.fit(df.compute()))
+        assert_estimator_equal(a.fit(df), b.fit(df.compute().values))
+
+    def test_estimator_comparison(self):
+        a = dpp.PolynomialFeatures()
+        b = spp.PolynomialFeatures()
+
+        a.fit_transform(X).compute()
+        b.fit_transform(X.compute())
+        assert_estimator_equal(a, b)
```

Comment:
I think that this `assert_estimator_equal` can be moved down to `test_array_transform` and this test can be removed.